src/rtc.rs

```diff
@@ -0,0 +1,484 @@
+/*!  Interface to the real time clock */
+
+use crate::rcc::{APB1, BDCR, CR, CSR};
+use crate::stm32::{PWR, RTC};
+use rtcc::{Datelike, Hours, NaiveDate, NaiveDateTime, NaiveTime, Rtcc, Timelike};
+
+/// Invalid input error
+// #[derive(Clone, Copy)]
+#[derive(Debug)]
+pub enum Error {
+    InvalidInputData,
+}
+
+/// RTC clock input source
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum RTCSrc {
+    LSE = 0b01,
+    LSI = 0b10,
+    HSE = 0b11,
+}
+
+/// RTC configuration struct
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct cfgRTC {
+    src: RTCSrc,
+    prediv_s: u16,
+    prediv_a: u8,
+}
+
+/// LSI clock source with prescalers
+/// set clock frequency to 1 hz
+impl Default for cfgRTC {
+    fn default() -> Self {
+        cfgRTC {
+            src: RTCSrc::LSI,
+            prediv_s: 311,
+            prediv_a: 127,
+        }
+    }
+}
+
+impl cfgRTC {
+    pub fn new(source: RTCSrc, prescaler_sync: u16, prescaler_async: u8) -> cfgRTC {
+        let config = cfgRTC {
+            src: source,
+            prediv_s: prescaler_sync,
+            prediv_a: prescaler_async,
+        };
+        config
+    }
+}
+
+pub struct Rtc {
+    regs: RTC,
+}
+
+impl Rtc {
+    pub fn rtc(
+        regs: RTC,
+        apb1: &mut APB1,
+        bdcr: &mut BDCR,
+        cr: &mut CR,
+        csr: &mut CSR,
+        cfg_rtc: cfgRTC,
+    ) -> Self {
+        match &cfg_rtc.src {
+            RTCSrc::LSI => Rtc::enable_lsi(csr),
+            RTCSrc::HSE => Rtc::enable_hse(cr, false),
+            RTCSrc::LSE => Rtc::enable_lse(bdcr, false),
```

Comment:
That is correct, the reason for exposing `CSR` and `CR` is to allow clock source selection. The LSE is not included on the STM32F303 Discovery board, so I initially wanted to make the rtc module flexible and allow the user to choose. That being said, if I had to choose one for support, it would be the LSE. I agree with you.